Src/LiquidProjections.Testing/EnumerableExtensions.cs

```diff
@@ -19,10 +19,7 @@ public static IEnumerable<IList<T>> InBatchesOf<T>(this IEnumerable<T> items, in
                 }
             }
 
-            if (batch.Count != 0)
-            {
-                yield return batch;
-            }
+            yield return batch;

```

Comment:
True, but the `MemoryEventSource` needs this to be able to meet my new requirement that an event source should always send an empty set of transactions right after subscribing, if there are no transactions beyond the provided checkpoint